CONTRIBUTING.md

```diff
@@ -0,0 +1,101 @@
+started working on it so that they won't waste any of their free time by
+working on their own fix/enhancement for it.
+
+If there isn't an open issue discussion for a bug fix/enhancement you want
+to work on please check that there isn't a closed discussion for it because
+a previous developer of it didn't finish their work which you could finish.
+
+And if there is no existing issue discussion for what you want to work on
+please open a new issue for it and tell other devs here that your going to/
+have started working on it.
+
+## Enhancement/Suggestions ##
+If you want to make a suggestion, please use the *Feature Request* option when making a new issue. It will provide all the information needed to evaluate the suggestion or enhancement.\
+![image](https://user-images.githubusercontent.com/20030128/123638352-fc609400-d816-11eb-88ad-11030e041fc4.png)
+
+
+## Pull Requests ##
+
+When providing a pull request ensure the description field describes the major
+changes from the request. If your pull request relates to any existing
+issues, include those issues in the description. The pull request provides a default layout to help you fill out this information.
+
+#### Pull Request Statuses ####
+
+CorsixTH uses the following labels inside your Pull Request's title to denote its current progress.
+| Label | Description |
+| ----- | ----------- |
+| **[WIP]** | Work in Progress |
+| **[RFC]** | Request for Comments (aka Ready for Review) |
+| **[RDY]** | Ready |
+
+Please make sure you label and update your Pull Request accordingly. Alternatively, you can submit a **Draft Pull Request** as an alternative to the **[WIP]** label.
+
+
+#### CONTRIBUTING CODE: ####
+*NB: If you're using GitHub Desktop ignore the git steps and follow guidance from [GitHub Docs](https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/overview/getting-started-with-github-desktop)*\
+**First Time**
+1. Ensure you have a GitHub account (https://github.com/signup/free)
+2. Fork CorsixTH\CorsixTH (https://github.com/CorsixTH/CorsixTH/fork)
+3. Ensure you have a git client.  ([GitHub Desktop](http://desktop.github.com) [Interactive] | [Git](https://git-scm.com/downloads) [Shell])
+4. Clone your fork to your computer
+	- ``git clone https://github.com/mygithubuser/CorsixTH.git``
+5. Add upstream remote
+	- ``git remote add upstream https://github.com/CorsixTH/CorsixTH.git``
+
+**Every Time**
+1. Sync your master branch with the CorisxTH repository's master branch
+	- ``git fetch upstream``
+	- ``git rebase upstream/master``
+2. Make sure no one is already working on the issue you want to work on.
+3. Tell other developers that you've started/will start working on this issue
+by posting a comment in its existing issue discussion or if there's no existing
+discussion for it then please open a new issue discussion for it and tell other
+devs here that your working on it.
+4. Create feature branch
+	- ``git branch myfeature upstream/master``
```

Comment:
You branch from your own master normally

``` shell
git checkout master
git branch myfeature
```
(and replace `myfeature` by a more descriptive line like `add-rats-levels`)

Reply:
Either is fine.
`git checkout -b myfeature master`